tests/test_types.py

```diff
@@ -3494,6 +3494,44 @@ class Model(BaseModel):
     }
 
 
+def test_path_like_extra_subtype():
+    class Model(BaseModel):
+        str_type: os.PathLike[str]
+        byte_type: os.PathLike[bytes]
+        any_type: os.PathLike[Any]
+
+    m = Model(
+        str_type='/foo/bar',
+        byte_type=b'/foo/bar',
+        any_type='/foo/bar',
+    )
+    assert m.str_type == Path('/foo/bar')
+    assert m.byte_type == Path('/foo/bar')
+    assert m.any_type == Path('/foo/bar')
+    assert Model.model_json_schema() == {
+        'properties': {
+            'str_type': {'format': 'path', 'title': 'Str Type', 'type': 'string'},
+            'byte_type': {'format': 'path', 'title': 'Byte Type', 'type': 'string'},
```

Comment:
Hmm, I wonder if we want to change the JSON schema for these path types...

Reply:
sure, what should we change them into ?

Reply:
I feel like the `type` should be `bytes` for the `byte_type` path...

Reply:
Honestly though, if someone complains we can fix this, I don't think it's super high priority and this gives us some flexibility to change it based on user experience.

Reply:
Bc it's really a string once you populate the path